src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -1143,6 +1143,9 @@ abstract class ClassfileParser {
         (newStub(name.toTypeName), newStub(name.toTermName))
       } else {
         val cls = owner.newClass(name.toTypeName, NoPosition, sflags)
+        if (cls.isProtected) {
```

Comment:
Looks like the right spot to me. The other cases seem to use `propagatePackageBoundary`. I'm not super familiar with this part. Assigning lrytz for review :-)